x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/saml/SamlAuthenticateRequest.java

```diff
@@ -47,4 +50,12 @@ public void setSaml(byte[] saml) {
     public void setValidRequestIds(List<String> validRequestIds) {
         this.validRequestIds = validRequestIds;
     }
+
+    public String getRealm() {
+        return realm;
+    }
+
+    public void setRealm(String realm) {
+        this.realm = realm;
+    }
```

Comment:
Erk! This class really ought to have serialization logic in it. Can we add that as a followup PR?
It's probably not the only SAML request/response that's missing it.

Reply:
will do